### ISSUE TITLE
Details elements aren't searchable - a11y issue

### ISSUE THREAD

**Issue by cassieevans** [facebook/docusaurus#10055]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

Hey folks.

The way that the details elements are implemented in docusaurus currently is incompatible with searching the page with cmd + F.

This is quite a big deal for documentation.

Closed details elements are searchable in chrome
https://chromestatus.com/feature/5032469667512320

I've tested in safari and firefox on my mac too and they also open on this page if I search for a word within
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details

However, in my site and on the docusaurus site they aren't searchable due to the styling that animates the height change.

searching for "correlation" on this page yields no results
https://docusaurus.io/docs/markdown-features

### Reproducible demo

https://docusaurus.io/docs/markdown-features

### Steps to reproduce

searching for "correlation" on this page yields no results 

### Expected behavior

the details element should open

### Actual behavior

it doesn't

### Your environment

- Public source code:
- Public site URL:
- Docusaurus version used:
- Environment name and version (e.g. Chrome 89, Node.js 16.4):
- Operating system and version (e.g. Ubuntu 20.04.2 LTS):

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by cassieevans** [comment]
Another little note - 

If I remove display none on the inner content div and then try and search the page - the browser finds the info and tries to toggle the details element open by adding an 'open' attribute.

Unfortunately the JS that runs this component seems to only change the "data-collapsed" attribute on click (maybe some other events too?)

A fix could be to listen to the onToggle event, then it would be aware that the browser was trying to open the details element.

**Comment by slorber** [comment]
Yes, I've also seen this problem. I thought it self-fixed somehow, because I saw it working recently, but apparently not 😅 

Surprisingly this is a "progressive degradation" because if you turn JS off, it's not possible to search those details elements.

I'm looking into it right now but it looks like this component should probably not "control" the open prop.

**Comment by cassieevans** [comment]
Thanks Sébastien!

**Comment by slorber** [comment]
Here's a first attempt, it works but introduces other bugs and also not sure the experience is great because the keyword does not highlight until the animation completes 😅 

https://github.com/facebook/docusaurus/pull/10057

This probably requires more work to figure this out 😅 

Maybe until we find a solution we could expose an option to disable that "animated progressive enhancement" through swizzle? This way your details won't be animated but at least they would be accessible.

**Comment by cassieevans** [comment]
Yeah - I think that would be a good middle ground in the interim. Thanks for looking into it. 🙏

Man this means I'm gonna have to update to v3 doesn't it 🫠 Been putting that one off because of the React 18/animation cleanup implications.

**Comment by cassieevans** [comment]
Ah, yeah that's a nice easy swizzle. I'll get on that. Thanks so much for the advice. Appreciated!

**Comment by cassieevans** [comment]
Got it all implemented with a little animated icon/animated expand  and it's working with search - thanks so much!

**Comment by slorber** [comment]
Note for myself, in 2025 we might be able to progressively enhance and animate details element without and JS:
https://nerdy.dev/open-and-close-transitions-for-the-details-element

I'm thinking of rebuilding a new Details component based on these new modern CSS attributes. We could expose it with an opt-in flag until browsers support those new CSS rules.

**Comment by taraspos** [comment]
Hi @slorber,
Was wondering if there are plans to fix this anytime soon, or https://github.com/facebook/docusaurus/issues/10055#issuecomment-2064108252 is the way to go at the moment?

**Comment by OfficialMutazAwad** [comment]
New reply to your ticket #13062

Title: Re: [facebook/docusaurus] Details elements aren&#039;t searchable - a11y issue (Issue #10055)

Message: Hello,we received your message and our support will reach out to you as soon as possible.

**Comment by slorber** [comment]
@taraspos for now swizzling is the way.

My plan is to build an entirely new details component that is more "native" and leverages newly released CSS features and HTML details improvements. 

Those features may not be widely supported by all browsers yet, so we may release this first behind a flag. It should be relatively safe to turn on anyway, as things like the height/arrow animations can be considered as progressive enhancements.

See for example: https://developer.chrome.com/docs/css-ui/animate-to-height-auto#animate_the_details_element

You can implement something like this today and show us the result. If this works well, we may steal your implementation, and it could become the "official" one. 

We may take the opportunity to also revisit the design, because I'm not sure it's good to have blue details that look a bit like our admonitions.

**Comment by lukaw3d** [comment]
Here's a swizzle wrapper workaround to patch it without reimplementing https://github.com/oasisprotocol/docs/pull/1569/files

Somewhat similar to https://github.com/facebook/docusaurus/pull/10057/files
